src/ima.js

```diff
@@ -289,6 +295,9 @@ export default class Ima extends BasePlugin {
       this._nextPromise = Utils.Object.defer();
       this._adDisplayContainer.initialize();
       this.player.ready().then(() => {
+        if (this._adsManager.isCustomPlaybackUsed()) {
+          this._contentSrc = this.player.getVideoElement().src;
```

Comment:
This won't be correct for playback with MSE (HLS/DASH), I guess you tested it on iOS with HLS.
But I guess this won't work at all with MSE in the first place as IMA will try to change the original source MSE and kill it anyway.

**Please make sure to test with HLS/DASH over desktop where we use MSE to see we didn't break it**.
The only place this could be problematic is chromecast ads but we still have milage till we get there.

Reply:
@OrenMe custom playback isn't true for desktop, It's usually for Safari/iOS and in this case we're playing native hls so it needs to be fine.
The only place we can have problem it's older versions of Android but if we take into account this is very rare I think this change is acceptable.

Reply:
@OrenMe maybe I can add to the condition && isIOS?

Reply:
> custom playback isn't true for desktop

This can change any day, but I agree for now.
So per my comment for testing - redundant now, and as you saw I already approved the PR previously ;-)

Reply:
> maybe I can add to the condition && isIOS?

I think we can address it when this is relevant - better abstract it in player level and allow re-setting a source but handling it by the underlying media source adapter.
The plugin doesn't need to have to know about internals of player playback.